pandas/core/util/hashing.py

```diff
@@ -76,7 +76,7 @@ def combine_hash_arrays(
         out ^= a
         out *= mult
         mult += np.uint64(82520 + inverse_i + inverse_i)
-    assert i + 1 == num_items, "Fed in wrong num_items"
+    assert len(arrays) == num_items, "Fed in wrong num_items"
```

Comment:
I'm not sure if `len` is capturing the same idea as `i + 1` here.

Reply:
This looks to me like a false positive.  If `arrays` is initially empty, then the method returns an empty array.  Otherwise, it's non-empty, which means that `i` will be defined.

Also, at this line, `arrays` is a chain, which apparently doesn't have `len`.

Reply:
Ah, okay. I'll comment out the original in-line.